src/models/transaction/TransferTransaction.ts

```diff
@@ -137,6 +152,40 @@ export class TransferTransaction extends Transaction {
     return this._mosaics!.map((_) => _.mosaicId);
   }
 
+  /**
+   * returns mosaic details without sorting through array
+   * @returns MosaicTransferable
+   */
+  public mosaicDetails = (): XEM | CACHE | MosaicTransferable => {
```

Comment:
Does XEM and CACHE inherit from MosaicTransferable? If so then we could just use polymorphism and do:
`public mosaicDetails = (): MosaicTransferable`